codelists/scripts/update_nhs_refsets.py

```diff
@@ -322,51 +327,67 @@ def run_bulk_import(cluster_file, config_file, host, release, live_run, names=No
         print(
             "  The suggestion is that if all the latest refsets were successfully imported above,"
         )
         print(
-            f"  then you should update the {RefsetVersionModel.__name__} record. If any failed, you may want to rerun"
+            f"  then you should update the {refset_version_model} record. If any failed, you may want to rerun"
         )
         print(
             "  this script but just for the failed refsets (retrying often works), and then update the"
         )
         print(
-            f"  {RefsetVersionModel.__name__} record once they have successfully imported.\n"
+            f"  {refset_version_model} record once they have successfully imported.\n"
         )
 
         update_confirm = (
             input(
-                f"Would you like to update the {RefsetVersionModel.__name__} record from {current_refset_tag} to {release['release_name']}? (y/n): "
+                f"Would you like to update the {refset_version_model} record from {latest_tag} to {release['release_name']}? (y/n): "
             )
             .strip()
             .lower()
         )
         if update_confirm in ("y", "yes"):
-            # Create a new version record
-            RefsetVersionModel.objects.create(
-                release=release["release"],
-                tag=release["release_name"],
-                release_date=release["valid_from"],
-            )
-            print(
-                f"Recorded new {RefsetVersionModel.__name__} version: {release['release_name']}"
-            )
+            # Update via API
+            update_url = f"{host}/coding-systems/update-refset-version/{refset_type}"
+            token = os.environ.get("API_TOKEN")
+            headers = {
+                "Authorization": f"Token {token}",
+                "Content-Type": "application/json",
+            }
+            payload = {
+                "release": release["release"],
+                "tag": release["release_name"],
+                "release_date": release["valid_from"].isoformat(),
+            }
+
+            try:
+                update_resp = requests.post(update_url, headers=headers, json=payload)
+                update_resp.raise_for_status()
+                result = update_resp.json()
+                print(
+                    f"✓ Successfully recorded new {refset_version_model} version: {release['release_name']}"
+                )
+                print(f"  Response: {result.get('message', 'Success')}")
+            except requests.exceptions.HTTPError as e:
+                print(f"Error updating refset version via API: {e}")
+                if e.response is not None:
+                    try:
+                        error_detail = e.response.json()
+                        print(f"  Error details: {error_detail}")
+                    except Exception:
+                        print(f"  Response: {e.response.text}")
```

Comment:
question: Is there any need to try and decode the JSON here if the request raises a 4xx or 5xx error? (If there is, then that's fine; just that it increasing code nesting?)